lib/jdomodels/src/test/java/org/sagebionetworks/repo/model/dbo/agent/AgentDaoImplTest.java

```diff
@@ -0,0 +1,198 @@
+		assertNotNull(session);
+
+		Thread.sleep(1001L);
+
+		// call under test
+		AgentSession updated = agentDao.updateSession(new UpdateAgentSessionRequest()
+				.setAgentAccessLevel(AgentAccessLevel.WRITE_YOUR_PRIVATE_DATA).setSessionId(session.getSessionId()));
+		assertNotNull(updated);
+
+		assertEquals(session.getSessionId(), updated.getSessionId());
+		assertEquals(session.getStartedBy(), updated.getStartedBy());
+		assertEquals(session.getAgentId(), updated.getAgentId());
+		assertEquals(session.getStartedOn(), updated.getStartedOn());
+		assertTrue(updated.getModifiedOn().getTime() > session.getStartedOn().getTime());
+		assertNotEquals(session.getEtag(), updated.getEtag());
+		assertEquals(AgentAccessLevel.WRITE_YOUR_PRIVATE_DATA, updated.getAgentAccessLevel());
+
+		assertEquals(Optional.of(updated), agentDao.getAgentSession(session.getSessionId()));
+	}
+
+	@Test
+	public void testMultipleSessions() {
+
+		AgentSession one = agentDao.createSession(adminUserId, new CreateAgentSessionRequest().setAgentId("one")
+				.setAgentAccessLevel(AgentAccessLevel.PUBLICLY_ACCESSIBLE));
+		AgentSession two = agentDao.createSession(adminUserId, new CreateAgentSessionRequest().setAgentId("two")
+				.setAgentAccessLevel(AgentAccessLevel.PUBLICLY_ACCESSIBLE));
+
+		assertNotEquals(one.getSessionId(), two.getSessionId());
+
+		AgentSession oneUp = agentDao.updateSession(new UpdateAgentSessionRequest().setSessionId(one.getSessionId())
+				.setAgentAccessLevel(AgentAccessLevel.READ_YOUR_PRIVATE_DATA));
+		AgentSession twoUp = agentDao.updateSession(new UpdateAgentSessionRequest().setSessionId(two.getSessionId())
+				.setAgentAccessLevel(AgentAccessLevel.WRITE_YOUR_PRIVATE_DATA));
+
+		assertEquals(AgentAccessLevel.READ_YOUR_PRIVATE_DATA, oneUp.getAgentAccessLevel());
+		assertEquals(AgentAccessLevel.WRITE_YOUR_PRIVATE_DATA, twoUp.getAgentAccessLevel());
+
+	}
+
+	@Test
+	public void testCreateSessionWithNullUserId() {
+		adminUserId = null;
+
+		String message = assertThrows(IllegalArgumentException.class, () -> {
+			// call under test
+			agentDao.createSession(adminUserId, createRequest);
+		}).getMessage();
+		assertEquals("userId is required.", message);
+	}
+
+	@Test
+	public void testCreateSessionWithNullRequest() {
+		createRequest = null;
+
+		String message = assertThrows(IllegalArgumentException.class, () -> {
+			// call under test
+			agentDao.createSession(adminUserId, createRequest);
+		}).getMessage();
+		assertEquals("request is required.", message);
+	}
+
+	@Test
+	public void testCreateSessionWithNullAccessLevel() {
+		createRequest.setAgentAccessLevel(null);
+
+		String message = assertThrows(IllegalArgumentException.class, () -> {
+			// call under test
+			agentDao.createSession(adminUserId, createRequest);
+		}).getMessage();
+		assertEquals("request.accessLevel is required.", message);
+	}
+
+	@Test
+	public void testCreateSessionWithNullAgentId() {
+		createRequest.setAgentId(null);
```

Comment:
I think we should always record the agent id even if it is the default, just in case we need past history across stack.